sklearn/multioutput.py

```diff
@@ -64,13 +64,24 @@ def _partial_fit_estimator(
     return estimator
 
 
+def _available_if_base_estimator_has(attr):
```

Comment:
does it make sense for it to be 

```suggestion
def _available_if_base_estimator_has(attr, base_estimator=None):
```

and have it more readable for people who read the code?

Reply:
I don't know what you mean by this. Do you mean that the second param is the attribute name? or somehow the estimator?

Reply:
sorry, yes, I realized now that my comment wasn't clear at all. Yes, the attribute name.

Reply:
My sense is that this function is trying to cover two different distinct names for "base_estimator", which means we either pass in the name of "base_estimator" or we have the classes define their own checks. Something like:

```python
class _MultiOutputEstimator(...):
    ...
    def _check_attr(self, attr):
        return hasattr(self.estimator, attr) or all(
            hasattr(est, attr) for est in self.estimators_
        )

    @available_if(lambda self: self._check_attr("partial_fit"))
    def partial_fit(self, X, y, classes=None, sample_weight=None):
```

```python
class _BaseChain(...):
    def _check_attr(self, attr):
        return hasattr(self.base_estimator, attr) or all(
            hasattr(est, attr) for est in self.estimators_
        )

class ClassifierChain(...)
    @available_if(lambda self: self._check_attr("predict_proba"))
    def predict_proba(self, X):
```